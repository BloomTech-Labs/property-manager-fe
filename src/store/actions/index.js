// IMPORTS/INITIALIZATION =========================|
// ================================================|
import axios from 'axios';
import axiosAuth from '../../helpers/axiosAuth';
// ------------------------------------------------|
// AUTH ACTIONS ===================================|
// ================================================|
export const AUTH_REQUEST_START = 'AUTH_REQUEST_START';
export const AUTH_REQUEST_SUCCESS = 'AUTH_REQUEST_SUCCESS';
export const AUTH_REQUEST_FAIL = 'AUTH_REQUEST_FAIL';
// ------------------------------------------------|
// LOGIN / SIGNUP ---------------------------------|
export const auth = url => (email, password) => async dispatch => {
  dispatch({ type: AUTH_REQUEST_START });

  try {
    const res = await axios.post(url, {
      email,
      password
    });

    localStorage.setItem('token', res.data.token);

    dispatch({ type: AUTH_REQUEST_SUCCESS, payload: { token: res } });
  } catch (err) {
    console.error(err);
    dispatch({ type: AUTH_REQUEST_FAIL, payload: { errorMessage: err } });
  }
};
// ------------------------------------------------|
// PROPERTY ACTIONS ===============================|
// ================================================|
export const ADD_PROPERTY_START = 'ADD_PROPERTY_START';
export const ADD_PROPERTY_SUCCESS = 'ADD_PROPERTY_SUCCESS';
export const ADD_PROPERTY_FAIL = 'ADD_PROPERTY_FAIL';
export const GET_PROPERTY_START = 'GET_PROPERTY_START';
export const GET_PROPERTY_SUCCESS = 'GET_PROPERTY_SUCCESS';
export const GET_PROPERTY_FAIL = 'GET_PROPERTY_FAIL';
// ------------------------------------------------|
// CREATE A PROPERTY ------------------------------|
export const createProperty = url => property => async dispatch => {
  dispatch({ type: ADD_PROPERTY_START });
  console.log(property);
  try {
    const res = await axios.post(url, { ...property });

    dispatch({
      type: ADD_PROPERTY_SUCCESS,
      payload: {
        created: res
      }
    });
  } catch (err) {
    console.error(err);

    dispatch({ type: ADD_PROPERTY_FAIL, payload: { errorMessage: err } });
  }
};
// ------------------------------------------------|
// GET PROPERTIES ---------------------------------|
export const getProperties = url => async dispatch => {
  dispatch({ type: GET_PROPERTY_START });

  try {
    const res = await axiosAuth().get(url);

    dispatch({
      type: GET_PROPERTY_SUCCESS,
      payload: {
        properties: res.data
      }
    });
  } catch (err) {
    console.error(err);

    dispatch({ type: GET_PROPERTY_FAIL, payload: { errMsg: err } });
  }
};
// ------------------------------------------------|
// USER ACTIONS ===================================|
// ================================================|

// define your user actions here

// ------------------------------------------------|
