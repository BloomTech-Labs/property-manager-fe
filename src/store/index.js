import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import { getFirebase } from 'react-redux-firebase';
import reducer from './reducers';

const middleware =
  process.ENV === 'production'
    ? [thunk.withExtraArgument(getFirebase)]
    : [thunk.withExtraArgument(getFirebase), logger];

export const makeStore = state =>
  createStore(
    reducer,
    state,
    compose(
      applyMiddleware(...middleware),

      /* eslint-disable */
      window.__REDUX_DEVTOOLS_EXTENSION__
        ? window.__REDUX_DEVTOOLS_EXTENSION__()
        : f => f
    )
  );

export default makeStore();
