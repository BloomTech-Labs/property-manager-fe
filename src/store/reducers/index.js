import { combineReducers } from 'redux';
import authReducer from './authReducer';
import propReducer from './propReducer';
import getUserReducer from './getUserReducer';
import workOrderReducer from './workOrderReducer';

export default combineReducers({
  authReducer,
  propReducer,
  getUserReducer,
  workOrderReducer
});
