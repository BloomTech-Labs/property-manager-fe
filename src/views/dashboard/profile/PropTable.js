import React from 'react';
import {
  makeStyles,
  Table,
  TableBody,
  TableCell,
  TableHead,
  TableRow,
  Paper
} from '@material-ui/core';
import TableContainer from '@material-ui/core/TableContainer';

const useStyles = makeStyles({
  table: {
    minWidth: 650
  }
});
function createData(property, tenants, payments) {
  return { property, tenants, payments };
}
const rows = [createData('123 Ez Street', 2, 2)];

export default function PropTable() {
  const classes = useStyles();
  // const properties = this.props.propertyList;

  return (
    <TableContainer component={Paper}>
      <Table classname={classes.table} aria-label="Property Table">
        <TableHead>
          <TableRow>
            <TableCell>Properties</TableCell>
            <TableCell align="right">Tenants</TableCell>
            <TableCell align="right">Payments</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {rows.map(row => (
            <TableRow key={row.property}>
              <TableCell component="th" scope="row">
                {row.property}
              </TableCell>
              <TableCell align="right">{row.tenants}</TableCell>
              <TableCell align="right">{row.payments}</TableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </TableContainer>
  );
}
