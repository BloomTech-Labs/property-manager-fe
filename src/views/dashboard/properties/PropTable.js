import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import {
  Table,
  TableBody,
  TableCell,
  TableHead,
  TableRow
} from '@material-ui/core';
import EditIcon from '@material-ui/icons/Edit';
import { navigate } from '@reach/router';
import useStyles from './propTableStyles';
import PropertyDetailsModal from '../../../components/Properties/PropertyDetailsModal';

export default function PropTable() {
  const classes = useStyles();

  const propertyList = useSelector(state => state.propReducer.properties);

  // open/close state for modal
  const [openDetails, setOpenDetails] = useState(false);

  // store individual property from map function in local state
  const [currentProperty, setCurrentProperty] = useState({});

  // handle open/close, takes in the individual property
  // passed up from property card to gain access from the modal
  const handleOpen = property => {
    // sets the currentProperty state to the property object passed in
    setCurrentProperty(property);

    // this sets the open state for the PropertyDetailsModal
    setOpenDetails(true);
  };

  const handleClose = () => {
    // sets currentProperty to empty obj when modal is closed
    setCurrentProperty({});

    // sets the open state to false to close PropertyDetailsModal
    setOpenDetails(false);
  };

  return (
    <>
      <Table className={classes.table} aria-label="Property Table">
        <TableHead>
          <TableRow>
            <TableCell>Id</TableCell>
            <TableCell align="right">Name</TableCell>
            <TableCell align="right">Street</TableCell>
            <TableCell align="right">State</TableCell>
            <TableCell align="right">No. Tenants</TableCell>
            <TableCell align="right">Edit</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {propertyList.map(prop => (
            <TableRow
              key={prop.id}
              className={classes.row}
              onClick={() => {
                navigate(`/dashboard/properties/${prop.id}`);
                handleOpen(prop);
              }}
            >
              <TableCell>{prop.id}</TableCell>
              <TableCell align="right">{prop.name}</TableCell>
              <TableCell align="right">{prop.street_address}</TableCell>
              <TableCell align="right">{prop.state}</TableCell>
              <TableCell align="right">{prop.occupied}</TableCell>
              <TableCell align="right">
                <EditIcon
                  style={{ cursor: 'pointer' }}
                  onClick={e => {
                    e.stopPropagation();
                    navigate(`/dashboard/properties/edit/${prop.id}`);
                  }}
                />
              </TableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
      <PropertyDetailsModal
        property={currentProperty}
        open={openDetails}
        close={handleClose}
      />
    </>
  );
}
