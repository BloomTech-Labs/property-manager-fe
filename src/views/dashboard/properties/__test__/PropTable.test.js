import React from 'react';
import * as rtl from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import store from '../../../../store';
import PropTable from '../PropTable';

afterEach(rtl.cleanup);

test('snapshot', () => {
  const { container } = rtl.render(
    <Provider store={store}>
      <PropTable />
    </Provider>
  );
  expect(container.firstChild).toMatchSnapshot();
});

test('Renders with name column', () => {
  const wrapper = rtl.render(
    <Provider store={store}>
      <PropTable />
    </Provider>
  );
  expect(wrapper.getByText(/name/i)).not.toBeNull();
});
