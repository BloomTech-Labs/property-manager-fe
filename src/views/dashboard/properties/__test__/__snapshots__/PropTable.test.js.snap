// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<table
  aria-label="Property Table"
  class="MuiTable-root makeStyles-table-1"
>
  <thead
    class="MuiTableHead-root"
  >
    <tr
      class="MuiTableRow-root MuiTableRow-head"
    >
      <th
        class="MuiTableCell-root MuiTableCell-head"
        scope="col"
      >
        Id
      </th>
      <th
        class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
        scope="col"
      >
        Name
      </th>
      <th
        class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
        scope="col"
      >
        Street
      </th>
      <th
        class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
        scope="col"
      >
        State
      </th>
      <th
        class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
        scope="col"
      >
        No. Tenants
      </th>
      <th
        class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
        scope="col"
      >
        Edit
      </th>
    </tr>
  </thead>
  <tbody
    class="MuiTableBody-root"
  />
</table>
`;
